msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/SSPRTest.kt

```diff
@@ -62,34 +52,24 @@ class SSPRTest : NativeAuthPublicClientApplicationAbstractTest() {
     }
 
     /**
-     * Running with runBlocking to avoid default 10 second execution timeout.
+     * Verify email with email OTP first and then reset password (hero scenario 8 & 17, use case 3.1.1) - Test case 46
      */
     @Test
-    fun testSSPRSuccessSimple() = runBlocking {
-        var retryCount = 0
-        var shouldRetry = true
+    fun testSSPRSuccess() = runBlocking {
+        var result: ResetPasswordStartResult
+        var otp: String
 
-        while (shouldRetry) {
-            try {
-                val user = NativeAuthCredentialHelper.nativeAuthSSPRUsername
-                val result = application.resetPassword(user)
-                Assert.assertTrue(result is ResetPasswordStartResult.CodeRequired)
-                val otp = tempEmailApi.retrieveCodeFromInbox(user)
+        retryOperation {
+            runBlocking {
```

Comment:
"Running with runBlocking to avoid default 10 second execution timeout." Maybe this comment can stay here.